account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -28,38 +49,56 @@ public String home() {
 		// Updating accountBalance after debit
 		
 		
-		@PostMapping("/debit/{accountNumber}/{amount}")
-		public User debit(@PathVariable String accountNumber, @PathVariable String amount) {
+		@PostMapping("/debit/{accountNumber}")
+		public void debit(@PathVariable String accountNumber, @RequestBody String amount) {
```

Comment:
Rather than saying String here, why don't you use as below?

```
		public void debit(@PathVariable Long accountNumber, @RequestBody Long amount) {

```

Ref: https://www.baeldung.com/spring-validate-requestparam-pathvariable#requestParam

Reply:
?

Reply:
Can't pass Long from RequestBody.